### DISCUSSION TITLE
Pandoc markdown vs Quarto markdown?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#5875] [Q&A] [answered]
### Description

The Quarto guide (Guide > Authoring > [Markdown basics](https://quarto.org/docs/authoring/markdown-basics.html#overview)) mentions:

> Quarto is based on Pandoc and uses its **variation** of markdown as its underlying document syntax.

This raises the following question: How does the markdown supported by Quarto differ from [Pandoc's markdown](https://pandoc.org/MANUAL.html#pandocs-markdown)? Or, where is this difference documented? For, all the Quarto guide shows are

> examples of the most _commonly used_ markdown syntax

suggesting that there is much more than what the page shows. Where is it?

**Comment by mcanouil** [comment, marked as answer]
> > Quarto is based on Pandoc and uses its **variation** of markdown as its underlying document syntax.
> 
> This raises the following question: How does the markdown supported by Quarto differ from [Pandoc's markdown](https://pandoc.org/MANUAL.html#pandocs-markdown)? Or, where is this difference documented? For, all the Quarto guide shows are

first "its" refers to Pandoc, while the second "its" refers to Quarto.
Feel free to make a PR to improve the wording.

> suggesting that there is much more than what the page shows. Where is it?

See <https://pandoc.org/MANUAL.html#markdown-variants>.
Then and for example for GitHub-Flavoured Markdown, the specifications can be found at <https://github.github.com/gfm/>.

**Reply by dense-set** [reply]
So the syntax of Quarto's `.qmd` files uses "a mix" of all those formats, _i.e._, all the markdown variations listed here <https://pandoc.org/MANUAL.html#markdown-variants>?

I understand that Quarto can _convert_ `.qmd` to such variations of markdown, but what I am asking is the "markdown language" that is used in `.qmd` files themselves.

**Reply by mcanouil** [reply]
The variants are examples of Markdown supported by Pandoc as output formats.

There are "extension" that enable/disable syntax rules for "markdown".
This are all described in Pandoc documentation.

**Reply by dense-set** [reply]
Oh, now I seem to have understood: We have variations of John Gruber's Markdown, with Pandoc's Markdown being one of them, in which Quarto's `.qmd` are written.  
Correct?

Secondly, what precisely, along with extensions, is the format of `.qmd` files? From what I read on Pandoc's documentation, the `unnumbered` class for headings is available via the extension `header-attributes`, so it must be enabled, no?

**Reply by mcanouil** [reply]
> Oh, now I seem to have understood: We have variations of John Gruber's Markdown, with Pandoc's Markdown being one of them, in which Quarto's .qmd are written.
Correct?

Yes, it's correct.

> Secondly, what precisely, along with extensions, is the format of .qmd files? From what I read on Pandoc's documentation, the unnumbered class for headings is available via the extension header-attributes, so it must be enabled, no?

Do you mean the following? (see <https://quarto.org/docs/reference/formats/html.html#numbering>)

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr>
<td>

````qmd
---
title: "Quarto Playground"
format: html
number-sections: true
---

This is a playground for Quarto.

# Hello World

Lorem ipsum dolor sit amet.

## Code {.unnumbered}

```{r}
1 + 1
```

## Comment

This is a comment.

````

</td>
<td>
<img width="576" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/2c98004e-d635-4f7e-99a9-1100f4748188">
</td>
</tr>
</table>

**Reply by dense-set** [reply]
> Yes, it's correct.

Thanks!

> Do you mean the following?

Yes. So is the format for `.qmd` files actually `markdown+header_attributes+...`?

@mcanouil Can you say something on this?

**Reply by mcanouil** [reply]
There are a lot of processing specific to Quarto which are not defined by format/plugin.
It's Pandoc ways of working.

If you remove all Quarto feature, it's the Pandoc markdown variant.
